### ISSUE TITLE
[Feature Request] Notifications bar of backup and restore progress

### ISSUE THREAD

**Issue by Cyberavater** [machiav3lli/oandbackupx#275]
This will help the progress run swiftly in the background. The current "Don't battery optimization" method for this app isn't much effective. It doesn't work well if you're backing up an app with external data with 4GB+, progress gets abandoned in the middle if the screen is turned off or you're doing something else keeping it in the background.

**Comment by machiav3lli** [comment]
You're still on 4.0.0?

**Comment by Cyberavater** [comment]
No, not at all! Maybe you misunderstood what I meant? Notification like you see in most downloader, it helps the process keep running in the background without any problem till it's there

**Comment by machiav3lli** [comment]
On the "keeping it in the *foreground", there's already a foreground notification. On the other matter, I'll see what I can do.

**Comment by Cyberavater** [comment]
For further reference, I've added a screenshot. You should be able to find relevant code in any open-source music player or downloader, as they use notifications to keep playing in the background.
![Screenshot_20201030-231041](https://user-images.githubusercontent.com/32187927/103845175-b6c3d980-50c5-11eb-935e-6589cc711e1a.png)

**Comment by opusforlife2** [comment]
> there's already a foreground notification

@machiav3lli Where? I only see a notification appear when an app is backed up or there is an error. There is no ongoing progress notification.

**Comment by Jakeler** [comment]
@Cyberavater @opusforlife2 It is possible that you have disabled some notifications for the app through system settings? 
I even get 2 notifications when running a batch backup (latest version):
![Screenshot_20210125-131643](https://user-images.githubusercontent.com/13378765/105705814-fdf8f980-5f10-11eb-9fdc-2efe016e7e47.png)

**Comment by opusforlife2** [comment]
Yup. Just one app. For example, Firefox Nightly can take some time to be backed up. If you switch to other apps and run out of RAM by chance, your backup silently fails.

**Comment by machiav3lli** [comment]
@opusforlife2 itcs already implemented in the test builds in form of a snackbar

**Comment by opusforlife2** [comment]
I see. Does the snackbar prevent the app from being booted from RAM?

My issue isn't that there is no visual indicator of an ongoing backup (though that is also a minor cosmetic issue). The intent is to prevent Android from considering the app process as low priority and killing it for no reason. As far as I know, only a foreground notification can do that.

**Comment by machiav3lli** [comment]
The single backup/restore actions use CoroutineWork APIs, making them run safe for 10 Minutes each. So answering your question, no it's not the snackbar that stops the process from being killed, but the fact that it's based on CoroutineWork.

**Comment by opusforlife2** [comment]
Alright. Thank you. I'll see if the longest backup I encounter overruns that time limit or not.

**Comment by opusforlife2** [comment]
@machiav3lli I see no snackbar when backing up a single app, though I can see it is in the changelog for version 5.1.0.

**Comment by machiav3lli** [comment]
@opusforlife2 it's actually under the app sheet, as it needs an activity context to be created in. Anyway, in the coming version, there'll be a notification to acompany it.

**Comment by opusforlife2** [comment]
Alright. I ask because as it is currently, there is still no way to see if there is a single backup currently in progress. If you get distracted by something and forget whether or not a backup is going on, you could initiate a second redundant backup in parallel.

By the way, why does the app let you do that? If an app is being backed up, then its backup button should be disabled for the duration of the ongoing backup, no? Should I open a new issue for this?
